test/versioned/langchain/runnables.tap.js

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const helper = require('../../lib/agent_helper')
+// load the assertSegments assertion
+require('../../lib/metrics_helper')
+const { filterLangchainEvents, filterLangchainMessages } = require('./common')
+const { version: pkgVersion } = require('@langchain/core/package.json')
+
+const { beforeHook, afterEachHook, afterHook } = require('../openai/common')
+
+tap.test('Langchain instrumentation - runnable sequence', (t) => {
```

Comment:
nice work on the tests, and helpers 👏🏻 . You may want to add a test that asserts no events are created when not in a transaction. I also don't see a test that asserts the span name

Reply:
This is an example for span name with tools

```js
 t.test('should create span on successful tools create', (t) => {
    const { agent, tool, input } = t.context
    helper.runInTransaction(agent, async (tx) => {
      const result = await tool.call(input)
      t.ok(result)
      t.assertSegments(tx.trace.root, ['Llm/tool/Langchain/node-agent-test-tool'], { exact: false })
      tx.end()
      t.end()
    })
```

Reply:
not in a transaction test

```js
 t.test('should not create llm tool events when not in a transaction', async (t) => {
    const { tool, agent, input } = t.context
    await tool.call(input)

    const events = agent.customEventAggregator.events.toArray()
    t.equal(events.length, 0, 'should not create llm events')
  })

  t.test('should add llm attribute to transaction', (t) => {
    const { tool, agent, input } = t.context
    helper.runInTransaction(agent, async (tx) => {
      await tool.call(input)

      const attributes = tx.trace.attributes.get(DESTINATIONS.TRANS_EVENT)
      t.equal(attributes.llm, true)

      tx.end()
      t.end()
    })
  })
```

Reply:
I've added these two tests. And added tests to test for skipping instrumentation as well.